include/CLI/Validators.hpp

```diff
@@ -113,7 +115,13 @@ class Validator {
         non_modifying_ = no_modify;
         return *this;
     }
-
+    /// Specify the application index of a validator
+    Validator &application_index(int app_index) {
```

Comment:
Could a const copying version work? This currently cannot be applied inline without changing the validator.

Reply:
I suppose the same is true for any of the validator modification functions.  `active`, `name`,`description`,`non_modifying`.  I think we would want them all to have the same paradigm.  

It would add a number of overloads but might make the calling a little easier for many of the const objects, which require an extra CLI::Validator( ) around them.  I will make that change tomorrow morning.